leaderelection/context.go

```diff
@@ -168,6 +167,32 @@ func (b *standardBuilder) buildElector(ctx context.Context, la reconciler.Leader
 	return &runAll{les: electors}, nil
 }
 
+// electorConfig returns the resource lock type, an unique ID and a list of Buckets based on
+// the ComponentConfig the standarBuiler has.
+func (b *standardBuilder) electorConfig(queueName string) (string, string, []reconciler.Bucket, error) {
+	lock := b.lec.LockType
+	if lock == "" {
+		lock = defaultKnativeResourceLock
+	} else if !supportedResourceLocks.Has(lock) {
+		return "", "", nil, fmt.Errorf("unsupported resource lock type: %s", b.lec.LockType)
+	}
+
+	id := b.lec.Identity
```

Comment:
if we re-arrange the code, we can get make it much nicer

1. get id
2. switch lock {case a: return ... case b: return...}

Reply:
done